### ISSUE TITLE
[Infra] Release process: JIRA release ticket is created even if marketplace deployment fails

### ISSUE THREAD

**Issue by duncanp-sonar** [SonarSource/sonarlint-visualstudio#4250]
### Description

The release pipeline is currently configured so that the JIRA release ticket is created in parallel with the artefacts being published to the VS marketplace and GitHub. This means the release ticket will be created to indicate that the technical release was done, even if one or more of the artefacts was not published.

This is not ideal as it has led to the situation where the PM notified the community that a new version has been released when in fact one or more artefacts are not available.

[MS docs](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/stages?view=azure-devops&tabs=classic)
[Release pipeline](https://dev.azure.com/sonarsource/DotNetTeam%20Project/_releaseDefinition?definitionId=13&_a=environments-editor-preview)

#### Current configuration - release ticket created in parallel
![image](https://github.com/SonarSource/sonarlint-visualstudio/assets/10757559/8b5c73da-033e-43ea-9790-954885214c93)

#### Note
The pipeline currently creates the JIRA release ticket and immediately marks it as `completed`. In theory,  we could create the release ticket and set the state as `started` when at the beginning of the pipeline and update it to `completed` once the pipeline has completed successfully. In fact, that is what the pipeline used to do, but it was broken when the state machine associated with the release ticket state was modified by another team. Fixing this is lower value, and would be a separate ticket.
